custom_components/waste_collection_schedule/waste_collection_schedule/source/avl_ludwigsburg.py

```diff
@@ -0,0 +1,143 @@
+import requests
+import json
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+from waste_collection_schedule.service.ICS import ICS
+from bs4 import BeautifulSoup
+
+TITLE = "avl-ludwigsburg.de"
+DESCRIPTION = "Abfallverwertungsgesellschaft des Landkreises Ludwigsburg mbH"
+URL = "https://www.avl-ludwigsburg.de/privatkunden/termine/abfallkalender/suche/"
+
+TEST_CASES = {
+    "CityWithoutStreet": {
+        "city": "Möglingen",
+    },
+    "CityWithStreet": {
+        "city": "Ludwigsburg",
+        "street": "Bahnhofstraße",
+    },
+}
+
+class Source:
+    def __init__(self, city, street=None):
+        self._city = city
+        self._street = street
+        self._ics = ICS()
+
+    def fetch(self):
+        # Get the hidden parameters by loading the page
+        s = requests.Session()
+        response = s.get(URL)
+        soup = BeautifulSoup(response.text, features="html.parser")
+        hidden_tags = soup.find_all("input", type="hidden")
+        
+        # Prepare data for the real web request
+        data = {}
+        for tag in hidden_tags:
+            data[tag.get("name")] = tag.get("value")
+        
+        # Find the cities which do need a street name
+        data_cities_with_streets = soup.find_all("input", type="text", placeholder="Ort eingeben")
+        cities_with_streets = ""
+        for tag in data_cities_with_streets:
+            cities_with_streets += tag.get("data-cities-with-streets")
+        cities_with_streets = cities_with_streets.split(",")
+        
+        data['tx_avlcollections_pi5[wasteCalendarLocationItem]']= self._city
+        data['tx_avlcollections_pi5[wasteCalendarStreetItem]']= self._street
+        
+        # Remove some data which the webserver doesn't like
+        data.pop('id', None)
+        data.pop('tx_kesearch_pi1[page]', None)
+        data.pop('tx_kesearch_pi1[resetFilters]', None)
+        data.pop('tx_kesearch_pi1[sortByField]', None)
+        data.pop('tx_kesearch_pi1[sortByDir]', None)
+
+        # Depending on the city remove the street from the data set
+        if self._city.lower() not in cities_with_streets:
+            data.pop('tx_avlcollections_pi5[wasteCalendarStreetItem]', None)
+
+        # Creat the headers and cookies
+        headers = {
+            'Connection': 'keep-alive',
```

Comment:
Do you really need all these header? Please try to use only the absolute minimum.

Reply:
Fixed with 95428a33a4c24c68deb81e84c3c33d251a3cb123